src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py

```diff
@@ -0,0 +1,225 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+
+from msrestazure.azure_exceptions import CloudError
+from azure.cli.core._util import CLIError
+from azure.cli.core.commands.arm import is_valid_resource_id
+from ._client_factory import (get_devtestlabs_management_client)
+from azure.mgmt.devtestlabs.models.gallery_image_reference import GalleryImageReference
+import azure.cli.core.azlogging as azlogging
+
+logger = azlogging.get_az_logger(__name__)
+devtestlabs_management_client = get_devtestlabs_management_client(None)
+
+
+def get_complex_argument_processor(expanded_arguments, assigned_arg, model_type):
+    """
+    Return a validator which will aggregate multiple arguments to one complex argument.
+    """
+    def _expension_valiator_impl(namespace):
+        """
+        The validator create a argument of a given type from a specific set of arguments from CLI
+        command.
+        :param namespace: The argparse namespace represents the CLI arguments.
+        :return: The argument of specific type.
+        """
+        ns = vars(namespace)
+        kwargs = dict((k, ns[k]) for k in ns if k in set(expanded_arguments))
+
+        setattr(namespace, assigned_arg, model_type(**kwargs))
+
+    return _expension_valiator_impl
+
+
+def validate_location(namespace):
+    """
+    Selects the default location of the lab when location is not provided while creating vm.
+    """
+    if namespace.location is None:
+        try:
+            lab_operation = devtestlabs_management_client.lab
+            lab = lab_operation.get_resource(namespace.resource_group, namespace.lab_name)
+            namespace.location = lab.location
+        except CloudError as err:
+            raise CLIError(err)
+
+
+def validate_network_parameters(namespace):
+    """
+    Selects the DevTest Lab's virtual network and subnet if not provided and updates namespace
+    """
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    vnet_operation = devtestlabs_management_client.virtual_network
+
+    if not namespace.vnet_name:
+        try:
+            lab_vnets = list(vnet_operation.list(namespace.resource_group, namespace.lab_name, top=1))
+            if not lab_vnets:
+                err = "Unable to find any virtual network in the '{}' lab.".format(namespace.lab_name)
+                raise CLIError(err)
+            else:
+                lab_vnet = lab_vnets[0]
+                namespace.vnet_name = lab_vnet.name
+                namespace.lab_virtual_network_id = lab_vnet.id
+        except CloudError as err:
+            raise CLIError(err)
```

Comment:
This looks like code solely in place for debugging.  CloudErrors should already be wrapped in CLIError, and thus you could remove this.